frontend/src/components/Home.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import { DigitLayout } from "@cthit/react-digit-components";
+import NewWindow from "react-new-window";
+import ReactDOM from "react-dom";
+import RecipeGridView from "./elements/home/RecipeGridView.jsx";
+import Recipe from "./Recipe.jsx";
+import EditRecipe from "./Edit.jsx";
+import axios from "axios";
+import _ from "lodash";
+import "./elements/home/styles/Home.css";
+
+class Home extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      recipes: []
+    };
+  }
+
+  componentDidMount() {
+    axios
+      .get("http://localhost:4000/getAllRecipes")
+      .then(res => {
+        console.log(res.data);
+        this.setState({
+          recipes: res.data
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  isUserCreator = creator => {
+    /*
+      Checks if the creator of recipe is the current user, currently
+      hardcoded
+    */
+    return true;
+  };
+
+  handleMenu = choice => recipe => {
+    if (choice === "edit_recipe") {
+      localStorage.setItem("recipeData", recipe);
+      // Route to edit page
+      window.open("/edit", "_self");
```

Comment:
`window.open` should not be used. Instead history.push from react-router-dom should be used. I think that `window.open` will refresh the entire page

Reply:
This applies to everywhere you're using window.open